packages/mui-joy/src/Breadcrumbs/breadcrumbsClasses.ts

```diff
@@ -1,4 +1,7 @@
-import { generateUtilityClass, generateUtilityClasses } from '@mui/base';
+import {
+  unstable_generateUtilityClass as generateUtilityClass,
+  unstable_generateUtilityClasses as generateUtilityClasses,
+} from '@mui/utils';
 
```

Comment:
**Off-topic**

I don't fully remember: what's the long-term goal with`@mui/utils` and how do we decide which modules it owns? From what I understand, this package is used by all the others, it's a shared kitchen sink. I'm asking about it because it feels like many of the modules it has been used directly by Joy UI, Material UI, and MUI X, without going through MUI Base.

One example: 

```jsx
import { unstable_useId as useId } from '@mui/utils';
```

https://github.com/mui/mui-x/blob/3e7bfb2a3b11d3728892b245e70928c7669a0bd6/packages/grid/x-data-grid/src/components/columnHeaders/GridColumnGroupHeader.tsx#L2

which makes me wonder if it would make sense to:

1. treat @mui/utils as an alpha unstable package, private APIs
2. reexport everything that is relevant in "@mui/base".
3. import from "@mui/base" wherever possible.
4. have @mui/utils host all the logic that could be required by @mui/base, @mui/system

in order to better sell the value proposition of MUI Base as a set of baseline helpers to create React components, including React utils.

Reply:
I don't remember us deciding on the shape of this library. To me, it should be a private package (I wouldn't say "alpha" or "unstable" - it's just internal).
As for re-exporting from MUI Base - I generally agree, but only the utilities that we intend to make public. I think Material UI and Joy could still import directly from utils. 

In the case of this PR, I changed the imports of `generateUtilityClass`, so that MUI Base could override the generic implementation from `@mui/utils` and add its own product name. As we ended up not using product classes, after all, this may not be necessary.

We can discuss it in the next meeting.

Reply:
> To me, it should be a private package (I wouldn't say "alpha" or "unstable" - it's just internal).

I'm personally OK with this. 

> I think Material UI and Joy could still import directly from utils.

Developers use the source of the components as a reference for how to create new ones. So if we bypass @mui/base, I think that it would mean that @mui/utils is not an internal help for @mui/base and @mui/system. But why not, we could imagine that @mui/utils is also part of MUI Base, and documented there too.